Ubuntu 18.04, gRPC v1.18.0, icu60でresembla_server.cppのmakeに失敗する

WikiのUsing Resembla via gRPCではgRPC v1.2.5を使用する例が掲載されていますが、(自分の理解では)Ubuntu 18.04ではOpenSSLの1.1が提供される関係上v1.2.5のビルドができないので、現行のv1.18.0とlibprotoc 3.6.1を使用したところ、resembla_server.cppのmakeに失敗します。
- Ubuntu 18.04
- MeCab 0.996-5
- libicu60, icu-devtools 60.2-3ubuntu3
- libsvm3, libsvm-dev 3.21+ds-1.1

~~~
rm -f /usr/local/lib/libproto* /usr/local/lib/libgrpc*

# protobufのmake, install
git clone -b $(curl -L https://grpc.io/release) https://github.com/grpc/grpc
cd grpc
git submodule update --init
cd third_party/protobuf
./autogen.sh
./configure
make
sudo make install

# gRPCのmake, install
cd ..
make
sudo make install

cd ../resembla/src/
make
sudo make install  # 成功する
# build resembla_index
cd executable
make
sudo make install  # 成功する
cd ../example/grpc/
./generate_grpc_templates.sh
cd src
make  # エラー
~~~

エラーの内容は
~~~
resembla_server.o: In function `RunServer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<resembla::ResemblaWithId<int> >, std::shared_ptr<resembla::StringNormalizer>, unsigned long, double)':
resembla_server.cpp:(.text+0x1068): undefined reference to `grpc::ServerBuilder::ServerBuilder()'
resembla_server.cpp:(.text+0x1075): undefined reference to `grpc::InsecureServerCredentials()'
resembla_server.cpp:(.text+0x1085): undefined reference to `grpc::ServerBuilder::AddListeningPort(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<grpc::ServerCredentials>, int*)'
resembla_server.cpp:(.text+0x109f): undefined reference to `grpc::ServerBuilder::RegisterService(grpc::Service*)'
resembla_server.cpp:(.text+0x10aa): undefined reference to `grpc::ServerBuilder::BuildAndStart()'
resembla_server.cpp:(.text+0x1137): undefined reference to `grpc::ServerBuilder::~ServerBuilder()'
resembla_server.cpp:(.text+0x1341): undefined reference to `grpc::ServerBuilder::~ServerBuilder()'
resembla_server.o: In function `std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > resembla::StringNormalizer::operator()<std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > >(std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > const&) const':
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x40): undefined reference to `vtable for icu_60::UnicodeString'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x5d): undefined reference to `void resembla::cast_string<std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >, icu_60::UnicodeString>(std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > const&, icu_60::UnicodeString&)'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x97): undefined reference to `icu_60::UnicodeString::moveFrom(icu_60::UnicodeString&)'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x9f): undefined reference to `icu_60::UnicodeString::~UnicodeString()'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0xee): undefined reference to `icu_60::UnicodeString::moveFrom(icu_60::UnicodeString&)'
resembla_server.o: In function `RunServer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<resembla::ResemblaWithId<int> >, std::shared_ptr<resembla::StringNormalizer>, unsigned long, double)':
resembla_server.cpp:(.text+0x1068): undefined reference to `grpc::ServerBuilder::ServerBuilder()'
resembla_server.cpp:(.text+0x1075): undefined reference to `grpc::InsecureServerCredentials()'
resembla_server.cpp:(.text+0x1085): undefined reference to `grpc::ServerBuilder::AddListeningPort(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<grpc::ServerCredentials>, int*)'
resembla_server.cpp:(.text+0x109f): undefined reference to `grpc::ServerBuilder::RegisterService(grpc::Service*)'
resembla_server.cpp:(.text+0x10aa): undefined reference to `grpc::ServerBuilder::BuildAndStart()'
resembla_server.cpp:(.text+0x1137): undefined reference to `grpc::ServerBuilder::~ServerBuilder()'
resembla_server.cpp:(.text+0x1341): undefined reference to `grpc::ServerBuilder::~ServerBuilder()'
resembla_server.o: In function `std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > resembla::StringNormalizer::operator()<std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > >(std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > const&) const':
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x40): undefined reference to `vtable for icu_60::UnicodeString'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x5d): undefined reference to `void resembla::cast_string<std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >, icu_60::UnicodeString>(std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > const&, icu_60::UnicodeString&)'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x97): undefined reference to `icu_60::UnicodeString::moveFrom(icu_60::UnicodeString&)'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0x9f): undefined reference to `icu_60::UnicodeString::~UnicodeString()'
resembla_server.cpp:(.text._ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_[_ZNK8resembla16StringNormalizerclINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEEET_RKS8_]+0xee): undefined reference to `icu_60::UnicodeString::moveFrom(icu_60::UnicodeString&)'
(以下、約800行)
Makefile:41: recipe for target 'resembla_server' failed
~~~

--- a/example/grpc/src/Makefile
+++ b/example/grpc/src/Makefile
@@ -21,7 +21,8 @@ all: $(BIN)
 
 CXX := g++
 CXXFLAGS := -Wall -Wextra -O3 -std=c++11 `pkg-config --cflags grpc++ grpc` -isystem../include -I../../../src -isystem../../../include -isystem../../../include/json -isystem../../../include/cmdline -isystem../../../include/paramset
-CXXLIBS := `pkg-config --libs icu-uc icu-i18n` `pkg-config --libs protobuf grpc++ grpc` -lgrpc++_reflection -lpthread -ldl -lresembla
+CXXLIBS := -lresembla `pkg-config --libs icu-uc icu-i18n` `pkg-config --libs protobuf grpc++ grpc` -lgrpc++_reflection -ldl -lsvm `mecab-config --libs`
+
 
 SUBDIRS = grpc
 
@@ -38,7 +39,7 @@ _grpc:
 	$(MAKE) -C grpc
 
 resembla_server: resembla_server.o _grpc
-	$(CXX) $(CXXLIBS) -o $@ resembla_server.o $(RESEMBLA_GRPC_OBJS)
+	$(CXX) resembla_server.o $(RESEMBLA_GRPC_OBJS) $(CXXLIBS) -o $@
 
 
 clean: $(SUBDIRS)

